client/resource_group/controller/controller.go

```diff
@@ -97,6 +97,9 @@ type ResourceGroupsController struct {
 
 	calculators []ResourceCalculator
 
+	mutex           sync.Mutex          // used for the `resourceCounter`
```

Comment:
What about using `sync.Map` directly?

Reply:
It says 
```
// The Map type is optimized for two common use cases: (1) when the entry for a given
// key is only ever written once but read many times, as in caches that only grow,
// or (2) when multiple goroutines read, write, and overwrite entries for disjoint
// sets of keys. In these two cases, use of a Map may significantly reduce lock
// contention compared to a Go map paired with a separate Mutex or RWMutex.
```
Here doesn't fit the optimized use case, so I just use the go map

Reply:
I think the write/read is in each group, not in the map level(groups). 
BTW, it better moves the Counter to `groupCostController`.

Reply:
Then, should we still use go map for storeCounter?